client/src/components/pages/Players.js

```diff
@@ -1,289 +1,320 @@
     } catch (e) {
-      alert(`Couldn't update team: ${e}`);
+      message.error(`Couldn't update team: ${e}`);
     }
-    this.setState({
-      loading: false,
-      editingTeam: -1,
-    });
+
+    setLoading(false);
+    setEditingTeam(-1);
   };
 
-  handleTeamEditStats = async (formData, _id) => {
-    const newTeam = await post("/api/team-stats", {
-      ...formData,
-      _id,
-      tourney: this.props.tourney,
-    });
-    this.setState((state) => ({
-      teams: state.teams.map((t) => {
+  const handleTeamEditStats = async (formData, _id) => {
+    try {
+      const newTeam = await post("/api/team-stats", {
+        ...formData,
+        _id,
+        tourney: tourney,
+      });
+
+      setTeams((t) => {
         if (t._id === _id) return newTeam;
+
         return t;
-      }),
-    }));
+      })
+    } catch {
+      message.error(`Couldn't get team stats: ${e}`);
+    }
   };
 
-  handlePlayerEdit = async (formData, _id) => {
-    const newPlayer = await post("/api/player-stats", {
-      ...formData,
-      _id,
-      tourney: this.props.tourney,
-      regTime: this.getStatsById(_id).regTime,
-    });
-    this.setState((state) => ({
-      players: state.players.map((t) => {
+  const handlePlayerEdit = async (formData, _id) => {
+    try {
+      const newPlayer = await post("/api/player-stats", {
+        ...formData,
+        _id,
+        tourney: tourney,
+        regTime: getStatsById(_id).regTime,
+      });
+
+      setPlayers(players.map((t) => {
         if (t._id === _id) return newPlayer;
+
         return t;
-      }),
-    }));
+      }));
+    } catch {
+      message.error(`Something went wrong: ${e}`);
+    }
   };
 
-  handleSortChange = (e) => {
+  const handleSortChange = (e) => {
     const sort = e.target.value;
-    if (this.state.mode === "players") {
-      this.sortPlayers(sort);
+    if (mode === "players") {
+      sortPlayers(sort);
     } else {
-      this.sortTeams(sort);
+      sortTeams(sort);
     }
 
-    this.setState({ sort });
+    setSort(sort);
   };
 
-  handleAddPlayer = async () => {
-    this.setState({ modalLoading: true });
-    const player = await post("/api/force-register", {
-      ...this.state.addPlayerData,
-      tourney: this.props.tourney,
-    });
-    this.setState(
-      (state) => ({
-        players: [...state.players, player],
-        modalLoading: false,
-        modalVisible: false,
-      }),
-      () => this.sortPlayers(this.state.sort)
-    );
+  const handleAddPlayer = async () => {
+    setModalLoading(true);
+
+    try {
+      const playerData = await post("/api/force-register", {
+        ...addPlayerData,
+        tourney: tourney,
+      });
+
+      setPlayers([...players, playerData]),
+      setModalLoading(false);
+      setModalVisible(false);
+
+      message.success(`${playerData.username} added!`);
+    } catch {
+      setModalLoading(false);
+      message.error('Player not found');
+    }
+
+    sortPlayers(sort);
```

Comment:
i think the reason the `setState` isn't working is because of this call to `sortPlayers`, which sorts the player list before the players state gets updated (since `setPlayers` is asynchronous)

this worked in the old version because `sortPlayers` was in the callback to the setstate, so it was guaranteed to be executed after the state was updated. however, we might need a different solution now.

i think one fix would be to have some function like `sortedPlayers(sort, players)` that accepts an array of players, and returns an array of sorted players instead of modifying the state directly
then, it could be used like `setPlayers(sortedPlayers(sort, [...players, playerData]))`

Reply:
Clear

Reply:
Oh yeah haven't done the prettier